snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrFirebaseMessagingService.java

```diff
@@ -80,10 +93,49 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
             data.put(key, value);
         }
 
+        PushTemplate template = processPushTemplate(data);
+        if (template != null){
+            // rich push, inject the template data into the context data we're passing down
+            data.put(SnapyrNotificationHandler.ACTION_BUTTONS_KEY, template);
+        }
+
         com.snapyr.sdk.Properties properties = new com.snapyr.sdk.Properties();
         properties.putAll(data);
         Snapyr.with(this).pushNotificationReceived(properties);
+        Snapyr.with(this).getNotificationHandler().showRemoteNotification(data);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
```

Comment:
Is there any rhyme or reason to the version of android that we're targeting?

Reply:
We were targeting `O` before only for one or two specific features of notifications that require that version - namely having a notification channel (category) on the notification. I don't think we need this check in general, only if there are certain things not supported by older versions (the action buttons maybe?)